src/arithmetic/entity_funcs/entity_funcs.c

```diff
@@ -49,19 +49,19 @@ SIValue AR_LABELS(SIValue *argv, int argc, void *private_data) {
 SIValue AR_HAS_LABELS(SIValue *argv, int argc, void *private_data) {
 	if(SI_TYPE(argv[0]) == T_NULL) return SI_NullVal();
 
-	bool         res       =  true;
-	Node         *node     =  argv[0].ptrval;
-	SIValue      labels    =  argv[1];
-	EntityID     id        =  ENTITY_GET_ID(node);
-	GraphContext *gc       =  QueryCtx_GetGraphCtx();
-	Graph        *g        =  gc->g;
+	bool         res    = true;
+	Node         *node  = argv[0].ptrval;
+	SIValue      labels = argv[1];
+	EntityID     id     = ENTITY_GET_ID(node);
+	GraphContext *gc    = QueryCtx_GetGraphCtx();
+	Graph        *g     = gc->g;
 
 	// iterate over given labels
 	uint32_t labels_length = SIArray_Length(labels);
 	for (uint32_t i = 0; i < labels_length; i++) {
 		SIValue label_value = SIArray_Get(labels, i);
-		if(SI_TYPE(label_value) != T_STRING) {
-			Error_SITypeMismatch(label_value, T_STRING);
+		if(!(SI_TYPE(label_value) & SI_STRING)) {
+			Error_SITypeMismatch(label_value, SI_STRING);
```

Comment:
revert Error_SITypeMismatch to report error on T_STRING no need to inform the user about intern string